tests/utils.rs

```diff
@@ -46,15 +81,22 @@ pub fn test_run_with_context(
     context: &Context,
```

Comment:
Do we still need `test_run_with_context` to be public? Or can we replace that with `test_run` everywhere, new that matching results is smarter?

Reply:
Yes, and we can also inline `test_run_with_context`.